### PR TITLE
CLOUDP-167771: events commands migration

### PR DESCRIPTION
**Pull request by wtrocki**
Migration for events commands

### CODE REVIEW

File: internal/cli/atlas/events/list.go [hunk 1]
```diff
@@ -24,7 +25,8 @@ import (
 	store "github.com/mongodb/mongodb-atlas-cli/internal/store/atlas"
 	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
 	"github.com/spf13/cobra"
-	atlas "go.mongodb.org/atlas/mongodbatlas"
+	"go.mongodb.org/atlas/mongodbatlasv2"
+	"k8s.io/utils/pointer"
```
Comment (gssbzn) [inline internal/cli/atlas/events/list.go:29]: wrong pointer package

File: internal/cli/atlas/events/orgs_list.go [hunk 2]
```diff
@@ -25,7 +26,8 @@ import (
 	store "github.com/mongodb/mongodb-atlas-cli/internal/store/atlas"
 	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
 	"github.com/spf13/cobra"
-	atlas "go.mongodb.org/atlas/mongodbatlas"
+	"go.mongodb.org/atlas/mongodbatlasv2"
+	"k8s.io/utils/pointer"
```
Comment (gssbzn) [inline internal/cli/atlas/events/orgs_list.go:30]: 👀

File: internal/cli/atlas/events/orgs_list.go [hunk 3]
```diff
@@ -44,11 +46,23 @@ func (opts *orgListOpts) initStore(ctx context.Context) func() error {
...
+	var r interface{}
 	var err error
-	r, err = opts.store.OrganizationEvents(opts.ConfigOrgID(), listOpts)
+	minDate, _ := time.Parse(time.RFC3339, opts.MinDate)
+	maxDate, _ := time.Parse(time.RFC3339, opts.MaxDate)
+
+	// TODO - event type is array but we expect single event
+	eventType, _ := mongodbatlasv2.NewEventTypeForOrgFromValue(opts.EventType[0])
+	listEventsAPIParams := mongodbatlasv2.ListOrganizationEventsApiParams{
+		OrgId:        opts.OrgID,
```
Comment (gssbzn) [inline internal/cli/atlas/events/orgs_list.go:57]: wrong orgID this won't leverage env variables or config file

File: internal/cli/atlas/events/orgs_list.go (hunk 2 above)
Comment (wtrocki) [reply internal/cli/atlas/events/orgs_list.go:30]: Forgot to disable Github copilot. It does keep introducing Kubernetes dependencies :(.

File: internal/cli/atlas/events/projects_list.go [hunk 4]
```diff
@@ -44,11 +46,25 @@ func (opts *projectListOpts) initStore(ctx context.Context) func() error {
...
-	var r *atlas.EventResponse
+	var r interface{}
 	var err error
-	r, err = opts.store.ProjectEvents(opts.ConfigProjectID(), listOpts)
+	minDate, _ := time.Parse(time.RFC3339, opts.MinDate)
+	maxDate, _ := time.Parse(time.RFC3339, opts.MaxDate)
+
+	eventType, _ := mongodbatlasv2.NewEventTypeForNdsGroupFromValue(opts.EventType[0])
+	listEventsAPIParams := mongodbatlasv2.ListProjectEventsApiParams{
+		GroupId:      opts.ProjectID,
```
Comment (gssbzn) [inline internal/cli/atlas/events/projects_list.go:56]: wrong project id, this wont' leverage env variables or config file

File: internal/cli/atlas/events/projects_list.go [hunk 5]
```diff
@@ -25,7 +26,8 @@ import (
 	store "github.com/mongodb/mongodb-atlas-cli/internal/store/atlas"
 	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
 	"github.com/spf13/cobra"
-	atlas "go.mongodb.org/atlas/mongodbatlas"
+	"go.mongodb.org/atlas/mongodbatlasv2"
+	"k8s.io/utils/pointer"
```
Comment (gssbzn) [inline internal/cli/atlas/events/projects_list.go:30]: 👀

File: internal/cli/atlas/events/orgs_list.go (hunk 3 above)
Comment (wtrocki) [reply internal/cli/atlas/events/orgs_list.go:57]: Good catch.

File: internal/store/atlas/events.go [hunk 6]
```diff
@@ -34,13 +34,57 @@ type EventLister interface {
 }
 
 // ProjectEvents encapsulate the logic to manage different cloud providers.
-func (s *Store) ProjectEvents(projectID string, opts *atlas.EventListOptions) (*atlas.EventResponse, error) {
-	result, _, err := s.client.Events.ListProjectEvents(s.ctx, projectID, opts)
+func (s *Store) ProjectEvents(opts *atlas.ListProjectEventsApiParams) (*atlas.GroupPaginatedEvent, error) {
+	request := s.clientv2.EventsApi.ListProjectEvents(s.ctx, opts.GroupId)
+	// TODO CLOUDP-173460 - enable using api params directly
```
Comment (wtrocki) [inline internal/store/atlas/events.go:39]: Fluent API problem

**Comment by wtrocki** [comment]
Because of only single event support, we have integration tests failing.

**Comment by matt-condon** [comment]
This is still blocked until we find a way to ensure the parameter array schema is included on the generated spec CLOUDP-173481

**Comment by matt-condon** [comment]
Closing this one and opening a separate PR as the branch has become a bit messy

**Pull request closed without merging**